18CO - Fix detection of the end of the President's Choice round on Ruby

ryandriskel opened the pull request:
Fix: https://github.com/tobymao/18xx/issues/3073
Fix: https://github.com/tobymao/18xx/issues/3094

Can this really be this simple? I feel like I saw :pass used elsewhere, but running the original sample game as a spec ( which has gone poof on 18xx.games ), the President's choice round never ended as the `@entities.delete(action.entity)` was never triggered because nothing matched :pass. Switching to a string worked in both ruby and browser.

tobymao commented on lib/engine/round/g_18_co/presidents_choice.rb:
```diff
@@ -23,9 +23,9 @@ def select_entities
+          return if action.type == 'message'
 
-          if action.type == :pass
+          if action.type == 'pass'
```
action.pass?

tobymao commented on lib/engine/round/g_18_co/presidents_choice.rb:
```diff
@@ -23,9 +23,9 @@ def select_entities
 
         def after_process(action)
-          return if action.type == :message
+          return if action.type == 'message'
```
action.free?

tobymao commented:
correct, opal doesn't have :symbol, ruby does. type is always a string, so when you do type == :x it is always false. but i think there are other ways you can handle your checks

ryandriskel commented:
I've incorporated Toby's suggestions and have the correct spec included this time. Somehow I had reverted to one from before the migrations and that's why the build was failing.
